src/components/AuthWrappers/PublicPageWrapper.tsx

```diff
@@ -23,9 +23,13 @@ export const PublicPageWrapper = ({
   const { hasLoginStateFlag } = useLoginState()
 
   if (hasLoginStateFlag && rest.strict) {
-    const callbackUrl =
-      rest.redirectUrl ?? String(router.query[CALLBACK_URL_KEY] ?? HOME)
-    void router.replace(callbackUrl)
+    if (rest.redirectUrl) {
+      void router.replace(callbackUrlSchema.parse(rest.redirectUrl))
+    } else {
+      void router.replace(
+        callbackUrlSchema.parse(router.query[CALLBACK_URL_KEY]),
+      )
+    }
```

Comment:
relooking at this again, would have probably been easier to replace it all with

```tsx
void router.replace(
  callbackUrlSchema.parse(rest.redirectUrl ?? router.query[CALLBACK_URL_KEY])
)
```

ah well